src/objects.c

```diff
@@ -1380,18 +1488,20 @@ void launch_new_connection(PgPool *pool, bool evict_if_needed)
 		}
 	}
 
-	max = user_max_connections(pool->user);
-	if (max > 0) {
-		/* try to evict unused connection first */
-		while (evict_if_needed && pool->user->connection_count >= max) {
-			if (!evict_user_connection(pool->user)) {
-				break;
+	if (pool->user) {
```

Comment:
this could use a comment, or an explicit boolean to make the code self-explanatory `if (!pool->is_peer)`

Reply:
Removed this check. I added dedicated pool_size checking for peer pools now and if that succeeds it skips over this section.